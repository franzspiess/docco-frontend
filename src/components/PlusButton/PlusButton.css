
.plus-button {
  border: thin solid #BDC3C7;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ECF0F1;
  transition: 500ms;
}

.plus-button:hover {
  transition: 500ms;
  border-color: #F1C40F;
  background-color: #F1C40F;
  transform: rotate(90deg);
}
