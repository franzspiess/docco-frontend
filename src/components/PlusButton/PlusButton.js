import React from 'react';
// import PropTypes from 'prop-types';
import './PlusButton.css';

const PlusButton = ({ click, size = 8 }) => {
  const myStyle = {
    width: `${size}vw`,
    height: `${size}vw`,
    fontSize: `${size}vw`,
    padding: `${size * 0.1}vw`
  };

  // PlusButton.propTypes = {
  //   size : PropTypes.number,
  //   click : PropTypes.func
  // }

  return (
    <div
      className="plus-button"
      style={myStyle}
      tabIndex="0"
      role="button"
      onClick={click}
      onKeyPress={click}
    >
      <i className="fas fa-plus" />
    </div>
  );
};

export default PlusButton;
