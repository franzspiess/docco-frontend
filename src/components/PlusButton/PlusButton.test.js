import React from 'react';
import { shallow } from 'enzyme';
import PlusButton from './PlusButton';

describe('<PlusButton />', () => {
  let wrapper;
  const mockClick = jest.fn();
  beforeEach(() => {wrapper = shallow(<PlusButton click={mockClick} />)});
  it('should render a div', () => {
    expect(wrapper.find('div')).toHaveLength(1);
  });
  it('should default its size to 8 if none it\'s passed', () => {
    expect(wrapper.find('div').prop('style').width).toBe('8vw');
  })
  it('should make use of font awesome', () => {
    expect(wrapper.find('.fa-plus')).toHaveLength(1);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a click prop containing a function that get\'s called on click', () => {
    expect(typeof wrapper.find('.plus-button').prop('onClick')).toBe('function');
    wrapper.find('.plus-button').simulate('click');
    expect(mockClick.mock.calls).toHaveLength(1);

  });

  //  couldn't get this test running but need to leave it in for further work;



});