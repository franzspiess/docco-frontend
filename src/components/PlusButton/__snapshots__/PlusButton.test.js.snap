// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlusButton /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton
    click={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="fas fa-plus"
      />,
      "className": "plus-button",
      "onClick": [MockFunction],
      "onKeyPress": [MockFunction],
      "role": "button",
      "style": Object {
        "fontSize": "8vw",
        "height": "8vw",
        "padding": "0.8vw",
        "width": "8vw",
      },
      "tabIndex": "0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fas fa-plus",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="fas fa-plus"
        />,
        "className": "plus-button",
        "onClick": [MockFunction],
        "onKeyPress": [MockFunction],
        "role": "button",
        "style": Object {
          "fontSize": "8vw",
          "height": "8vw",
          "padding": "0.8vw",
          "width": "8vw",
        },
        "tabIndex": "0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fas fa-plus",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
